Documentation/kubernetes/install/kubespray.rst

```diff
@@ -10,7 +10,176 @@
+Configure AWS credentials
+-------------------------
+
+Export the variables for your AWS credentials 
+
+.. code:: bash
+
+  export AWS_ACCESS_KEY_ID="www"
+  export AWS_SECRET_ACCESS_KEY ="xxx"
+  export AWS_SSH_KEY_NAME="yyy"
+  export AWS_DEFAULT_REGION="zzz"
+
+-----------------------------
+Configure Terraform Variables
+-----------------------------
+
+We will start by specifying the infrastructure needed for the Kubernetes cluster.
+
+.. code:: bash
+
+  $ cd contrib/terraform/aws
+  $ cp contrib/terraform/aws/terraform.tfvars.example terraform.tfvars`
+
+Open the file and change any defaults particularly, the number of master, etcd, and worker nodes. 
+You can change the master and etcd number to 1 for deployments that don't need high availability.
+By default, this tutorial will create:
+
+  - VPC with 2 public and private subnets
+  - Bastion Hosts and NAT Gateways in the Public Subnet
+  - Three of each (masters, etcd, and worker nodes) in the Private Subnet
+  - AWS ELB in the Public Subnet for accessing the Kubernetes API from
+    the internet
+  - Terraform scripts using ``CoreOS`` as base image.
+
+Example ``terraform.tfvars`` file:
+
+.. code:: bash
+
+  #Global Vars
+  aws_cluster_name = "kubespray"
+
+  #VPC Vars
+  aws_vpc_cidr_block = "XXX.XXX.192.0/18"
+  aws_cidr_subnets_private = ["XXX.XXX.192.0/20","XXX.XXX.208.0/20"]
+  aws_cidr_subnets_public = ["XXX.XXX.224.0/20","XXX.XXX.240.0/20"]
+
+  #Bastion Host
+  aws_bastion_size = "t2.medium"
+
+
+  #Kubernetes Cluster
+
+  aws_kube_master_num = 3
+  aws_kube_master_size = "t2.medium"
+
+  aws_etcd_num = 3
+  aws_etcd_size = "t2.medium"
+
+  aws_kube_worker_num = 3
+  aws_kube_worker_size = "t2.medium"
+
+  #Settings AWS ELB
+
+  aws_elb_api_port = 6443
+  k8s_secure_api_port = 6443
+  kube_insecure_apiserver_address = "0.0.0.0"
+
+
+-----------------------
+Apply the configuration
+-----------------------
+
+``terraform init`` to initialize the following modules
+
+  - ``module.aws-vpc``
+  - ``module.aws-elb``
+  - ``module.aws-iam``
+
+.. code:: bash
+
+  $ terraform init
+
+Once initialized , execute:
+
+.. code:: bash
+
+  $ terraform plan -out=aws_kubespray_plan
+
+This will generate a file, ``aws_kubespray_plan``, depicting an execution
+plan of the infrastructure that will be created on AWS. To apply, execute:
+
+.. code:: bash
+
+  $ terraform init
+  $ terraform apply "aws_kubespray_plan"
+
+Terraform automatically creates an Ansible Inventory file at ``inventory/hosts``.
+
+Installing Kubernetes cluster with Cilium as CNI
+================================================
+
+Kubespray uses Ansible as its substrate for provisioning and orchestration. Once the infrastructure is created, you can run the Ansible playbook to install Kubernetes and all the required dependencies. Execute the below command in the kubespray clone repo, providing the correct path of the AWS EC2 ssh private key in ``ansible_ssh_private_key_file=<path to EC2 SSH private key file>``
```

Comment:
Link to Ansible

Reply:
Not much use since if you have made the choice of kubespray less likely that you are unaware of Ansible...

Reply:
I'll defer to you on this since I don't know kubespray that well.